src/System.Runtime.Extensions/src/System/IO/Path.Unix.cs

```diff
@@ -43,23 +39,105 @@ private static bool IsDirectoryOrVolumeSeparator(char c)
 
         private static void EmulateFileIOPermissionChecks(string fullPath)
```

Comment:
Can we rename this method to make it more clear what's going on?  I would not expect "EmulateFileIOPermissionChecks" to be a no-op on Unix.  Alternately, pull GetFullPath into the PAL since that seems to be the only caller in the platform agnostic code?